src/app/models/itinenary.ts

```diff
@@ -0,0 +1,14 @@
+export interface Itinerary {
+  itineraryId: string
+  itineraryName: string
+  itineraryStartTime: string
+  itineraryEndTime: string
+  tag:string
+  currency? : string
+  itineraryCost: number
+  itineraryStartLocation: string
+  itineraryEndLocation: string
+  notes: string
+  tripId:string;
+  id?:string;
```

Comment:
Rather make a partial type from this with `type NewItinerary = Omit<Itinerary, 'id'>` if that's why you've made id optional. It's not optional to a DB.